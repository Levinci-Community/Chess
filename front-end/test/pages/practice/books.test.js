import React from "react";
import { render } from "@testing-library/react";
import BooksPage from "../../../src/pages/practice/books";
import { BrowserRouter as Router } from "react-router-dom";

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
      t: (key) => key,
    }),
  }));
  
  test("renders BooksPage snapshot", () => {
    const { container } = render(
      <Router>
        <BooksPage />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });