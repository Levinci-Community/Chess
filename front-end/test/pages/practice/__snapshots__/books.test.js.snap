// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BooksPage snapshot 1`] = `
<div>
  <div
    class="chakra-container css-3laa41"
  >
    <div
      class="css-gg4vpm"
    >
      <h2
        class="chakra-heading css-zx6l2c"
      >
        books.books
      </h2>
      <div
        class="css-k008qs"
      >
        <div
          class="css-1lxvmmi"
        >
          <input
            class="chakra-input css-1v5z18m"
            placeholder="common.search"
          />
          <button
            class="chakra-button css-1t6whxw"
            title="search"
            type="button"
          >
            <svg
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="css-k008qs"
    >
      <div
        class="css-uyoyg8"
      >
        <div
          class="css-9fzojd"
        >
          <div
            class="chakra-button__group css-1ind3v2"
            data-orientation="horizontal"
            role="group"
          >
            <button
              class="chakra-button css-1fysbuj"
              title="left"
              type="button"
            >
              <svg
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="chakra-button css-1fysbuj"
              title="right"
              type="button"
            >
              <svg
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="css-17xejub"
      />
      <div
        class="css-18467a"
      >
        <div
          class="chakra-container css-1fhgjcy"
        >
          <div
            class="chakra-stack css-1igwmid"
          >
            <div
              class="css-1v3hqhg"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 8px;"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M172.2 226.8c-14.6-2.9-28.2 8.9-28.2 23.8V301c0 10.2 7.1 18.4 16.7 22 18.2 6.8 31.3 24.4 31.3 45 0 26.5-21.5 48-48 48s-48-21.5-48-48V120c0-13.3-10.7-24-24-24H24c-13.3 0-24 10.7-24 24v248c0 89.5 82.1 160.2 175 140.7 54.4-11.4 98.3-55.4 109.7-109.7 17.4-82.9-37-157.2-112.5-172.2zM209 0c-9.2-.5-17 6.8-17 16v31.6c0 8.5 6.6 15.5 15 15.9 129.4 7 233.4 112 240.9 241.5.5 8.4 7.5 15 15.9 15h32.1c9.2 0 16.5-7.8 16-17C503.4 139.8 372.2 8.6 209 0zm.3 96c-9.3-.7-17.3 6.7-17.3 16.1v32.1c0 8.4 6.5 15.3 14.8 15.9 76.8 6.3 138 68.2 144.9 145.2.8 8.3 7.6 14.7 15.9 14.7h32.2c9.3 0 16.8-8 16.1-17.3-8.4-110.1-96.5-198.2-206.6-206.7z"
                />
              </svg>
              <p
                class="chakra-text css-15akpqp"
              >
                header.blogs
              </p>
            </div>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-adijlq"
            />
          </div>
          <ul
            class="css-1pncojb"
            role="list"
          />
          <div
            class="chakra-stack css-1igwmid"
          >
            <div
              class="css-1v3hqhg"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 8px;"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                />
              </svg>
              <p
                class="chakra-text css-15akpqp"
              >
                header.books
              </p>
            </div>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-adijlq"
            />
          </div>
          <ul
            class="css-1pncojb"
            role="list"
          />
          <div
            class="chakra-stack css-1igwmid"
          >
            <div
              class="css-1v3hqhg"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 8px;"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M336.2 64H47.8C21.4 64 0 85.4 0 111.8v288.4C0 426.6 21.4 448 47.8 448h288.4c26.4 0 47.8-21.4 47.8-47.8V111.8c0-26.4-21.4-47.8-47.8-47.8zm189.4 37.7L416 177.3v157.4l109.6 75.5c21.2 14.6 50.4-.3 50.4-25.8V127.5c0-25.4-29.1-40.4-50.4-25.8z"
                />
              </svg>
              <p
                class="chakra-text css-15akpqp"
              >
                header.videos
              </p>
            </div>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-adijlq"
            />
          </div>
          <ul
            class="css-1pncojb"
            role="list"
          />
          <div
            class="chakra-stack css-1igwmid"
          >
            <div
              class="css-1v3hqhg"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 8px;"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
                />
              </svg>
              <p
                class="chakra-text css-15akpqp"
              >
                header.play
              </p>
            </div>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-adijlq"
            />
          </div>
          <ul
            class="css-1pncojb"
            role="list"
          >
            <li
              class="css-8atqhb"
              item="[object Object]"
            >
              <a
                class="chakra-link css-1qqhvmo"
                href="/new-game"
              >
                header.play
              </a>
            </li>
            <li
              class="css-8atqhb"
              item="[object Object]"
            >
              <a
                class="chakra-link css-1qqhvmo"
                href="/lobby"
              >
                header.lobby
              </a>
            </li>
            <li
              class="css-8atqhb"
              item="[object Object]"
            >
              <a
                class="chakra-link css-1qqhvmo"
                href="/tournaments"
              >
                header.tournaments
              </a>
            </li>
          </ul>
          <div
            class="chakra-stack css-1igwmid"
          >
            <div
              class="css-1v3hqhg"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="margin-right: 8px;"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M519.442 288.651c-41.519 0-59.5 31.593-82.058 31.593C377.409 320.244 432 144 432 144s-196.288 80-196.288-3.297c0-35.827 36.288-46.25 36.288-85.985C272 19.216 243.885 0 210.539 0c-34.654 0-66.366 18.891-66.366 56.346 0 41.364 31.711 59.277 31.711 81.75C175.885 207.719 0 166.758 0 166.758v333.237s178.635 41.047 178.635-28.662c0-22.473-40-40.107-40-81.471 0-37.456 29.25-56.346 63.577-56.346 33.673 0 61.788 19.216 61.788 54.717 0 39.735-36.288 50.158-36.288 85.985 0 60.803 129.675 25.73 181.23 25.73 0 0-34.725-120.101 25.827-120.101 35.962 0 46.423 36.152 86.308 36.152C556.712 416 576 387.99 576 354.443c0-34.199-18.962-65.792-56.558-65.792z"
                />
              </svg>
              <p
                class="chakra-text css-15akpqp"
              >
                header.practice
              </p>
            </div>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-adijlq"
            />
          </div>
          <ul
            class="css-1pncojb"
            role="list"
          >
            <li
              class="css-8atqhb"
              item="[object Object]"
            >
              <a
                class="chakra-link css-1qqhvmo"
                href="/puzzles"
              >
                header.puzzles
              </a>
            </li>
            <li
              class="css-8atqhb"
              item="[object Object]"
            >
              <a
                class="chakra-link css-1qqhvmo"
                href="/books"
              >
                header.books
              </a>
            </li>
            <li
              class="css-8atqhb"
              item="[object Object]"
            >
              <a
                class="chakra-link css-1qqhvmo"
                href="videos"
              >
                header.videos
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
